### PR TITLE
Unify usage of /tmp folder and .tmp extension for writing temp files 

### PR DESCRIPTION
**Pull request by NihalHarish**
### Description of changes:
- Removed the usage of NON_SAGEMAKER_TEMP_PATH_PREFIX

#### Style and formatting:

I have run `pre-commit install` to ensure that auto-formatting happens with every commit.

#### Issue number, if available

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: smdebug/core/access_layer/file.py [hunk 1]
```diff
@@ -4,12 +4,10 @@
 
 # First Party
 from smdebug.core.logger import get_logger
-from smdebug.core.sagemaker_utils import is_sagemaker_job
 
 # Local
 from .base import TSAccessBase
 
-NON_SAGEMAKER_TEMP_PATH_PREFIX = "/tmp"
 SAGEMAKER_TEMP_PATH_SUFFIX = ".tmp"
```
Comment (rahul003) [inline smdebug/core/access_layer/file.py]: rename this variable too?

File: smdebug/core/utils.py [hunk 2]
```diff
@@ -84,11 +84,10 @@ def list_files_in_directory(directory, file_regex=None):
 
 
 def list_collection_files_in_directory(directory):
-    collections_directory = get_path_to_collections(directory)
     import re
 
-    collections_file_regex = re.compile(".*_?collections.json")
-    return list_files_in_directory(collections_directory, file_regex=collections_file_regex)
+    collections_file_regex = re.compile(".*_?collections.json$")
+    return list_files_in_directory(directory, file_regex=collections_file_regex)
```
Comment (rahul003) [inline smdebug/core/utils.py]: why did you remove the collections directory?

File: smdebug/core/utils.py [hunk 3]
```diff
@@ -84,11 +84,10 @@ def list_files_in_directory(directory, file_regex=None):
 
 
 def list_collection_files_in_directory(directory):
-    collections_directory = get_path_to_collections(directory)
     import re
 
-    collections_file_regex = re.compile(".*_?collections.json")
-    return list_files_in_directory(collections_directory, file_regex=collections_file_regex)
+    collections_file_regex = re.compile(".*_?collections.json$")
```
Comment (rahul003) [inline smdebug/core/utils.py]: What about the regexes for index and event files?

Comment (NihalHarish) [reply smdebug/core/utils.py]: Index and event files already used the correct regex patterns.

File: smdebug/core/utils.py (hunk 2 above)
Comment (NihalHarish) [reply smdebug/core/utils.py]: the function call is invoked like this, so there is no need get collections directory again
`return list_collection_files_in_directory(get_path_to_collections(self.path))`

File: smdebug/core/access_layer/file.py (hunk 1 above)
Comment (NihalHarish) [reply smdebug/core/access_layer/file.py]: renamed `SAGEMAKER_TEMP_PATH_SUFFIX` to `SMDEBUG_TEMP_PATH_SUFFIX`

File: smdebug/core/utils.py (hunk 2 above)
Comment (rahul003) [reply smdebug/core/utils.py]: Slightly confused as to how it was working earlier
